app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -132,22 +141,95 @@ class ReportingRepositoryImpl(
         tanUuid = apiInteractor.requestTan(mobileNumber).uuid
     }
 
-    override suspend fun uploadReportInformation(
-        temporaryExposureKeys: List<TemporaryExposureKey>
-    ): MessageType {
+    override suspend fun uploadReportInformation(): MessageType {
         return when (messageTypeSubject.value) {
-            MessageType.Revoke.Suspicion -> uploadRevokeSuspicionInfo(temporaryExposureKeys)
-            MessageType.Revoke.Sickness -> uploadRevokeSicknessInfo(temporaryExposureKeys)
-            else -> uploadInfectionInfo(temporaryExposureKeys)
+            MessageType.Revoke.Suspicion -> uploadRevokeSuspicionInfo()
+            MessageType.Revoke.Sickness -> uploadRevokeSicknessInfo()
+            else -> uploadInfectionInfo()
         }
     }
 
-    private suspend fun uploadInfectionInfo(
-        temporaryExposureKeys: List<TemporaryExposureKey>): MessageType.InfectionLevel {
+    private suspend fun uploadInfectionInfo(): MessageType.InfectionLevel {
         return withContext(coroutineContext) {
-            val infectionLevel = messageTypeSubject.value as? MessageType.InfectionLevel ?: throw InvalidConfigurationException.InfectionLevelNotSet
+            val infectionLevel = messageTypeSubject.value as? MessageType.InfectionLevel
+                ?: throw InvalidConfigurationException.InfectionLevelNotSet
+
+            val configuration = configurationRepository.observeConfiguration().blockingFirst()
+            val uploadKeysDays = configuration.uploadKeysDays
```

Comment:
we should not throw an exception here but provide a reasonable default. This looks like it would be an developer error it this value was null

Reply:
⬇

Reply:
I would avoid hardcoding defaults for different configurations, this will just hide the problem. If an issue that it's totally unexpected happens it's better to not about it and that's why it's thrown here.

Reply:
Let's keep it.